docs/data/data-grid/editing/editing.md

```diff
@@ -401,6 +401,56 @@ The following demo requires a value for the **Payment method** column only if th
 
 {{"demo": "ConditionalValidationGrid.js", "disableAd": true, "bg": "inline", "defaultCodeOpen": false}}
 
+### Linked fields
+
+The options available for one field may depend on the value of another field.
+For instance, if the `singleSelect` column is used, you can provide a function to `valueOptions` returning the relevant options for the value selected in another field, as exemplified below.
+
+```tsx
+const columns: GridColDef[] = [
+  {
+    field: 'account',
+    type: 'singleSelect',
+    valueOptions: ({ row }) => {
+      return row.type === 'Income' // Gets the value of the "type" field
```

Comment:
The row being undefined for the filter, maybe put an example that is compatible with an usage in the filters (to return the two arrays concatenated if row is undefined for instance)

Same on the full example at the end of the section, it crashes when filtering.

____

Side note: 

TS does not complain because in `GridColumns`, by default `R = any` and `any | undefined` is equal to `any`

If I change `R` default value to `{ [key: string]: any }` it does complain and you would have spotted the bug.
I think I put `any` because otherwise you can have type conflicts between typed and untyped `R`, I will try to spend time on it someday.

Reply:
Updated. For v6 we can change the default type of `R`.

Reply:
> I think I put any because otherwise you can have type conflicts between typed and untyped R, I will try to spend time on it someday.

Not sure, I did not do it because it was a mess
Maybe there is a way to do it properly but I did not find it